apps/explorer/src/ui/TableCard.tsx

```diff
@@ -46,29 +52,53 @@ type DataType = {
         | TxStatus;
 };
 
-function columnsContent(columns: TableColumn[]) {
+function columnsContent(columns: TableColumn[], sortTable: boolean) {
     return columns.map((column) => ({
-        accessorKey: column.accessorKey,
-        id: column.accessorKey,
         header: column.headerLabel,
+        ...column,
         // cell renderer for each column from react-table
-        cell: (info: any) => info.getValue(),
+        // cell should be in the column definition
+        //TODO: move cell to column definition
+        ...(!sortTable && { cell: (info: any) => info.getValue() }),
     }));
 }
 
 export interface TableCardProps {
     refetching?: boolean;
     data: DataType[];
     columns: TableColumn[];
+    sortTable?: boolean;
+    defaultSorting?: SortingState;
 }
 
-export function TableCard({ refetching, data, columns }: TableCardProps) {
+export function TableCard({
+    refetching,
+    data,
+    columns,
+    sortTable,
+    defaultSorting,
+}: TableCardProps) {
     // Use Columns to create a table
-    const processedcol = useMemo(() => columnsContent(columns), [columns]);
+    const processedcol = useMemo(
```

Comment:
we can improve typings here e.g. `useMemo<ColumnDef<TableData>[]> where `TableData` is a type representing the data in the table